### PR TITLE
feat: pprof

### PR DESCRIPTION
**Pull request by epociask**
# Pull Request Template

## Fixes Issue

Fixes #

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -34,7 +35,14 @@ const (
...
 )
 
+type SvrConfig struct {
+	Host string
+	Port int
+	PPROF bool
+}
+
 type Server struct {
+	pprof 	   bool
```
Comment (bxue-l2) [inline server/server.go:45]: embed SvrConfig within the server to avoid putting pprof in many places

**Review by bxue-l2 — APPROVED** [review]

File: e2e/setup.go [hunk 2]
```diff
@@ -123,7 +123,13 @@ func CreateTestSuite(t *testing.T, useMemory bool, useS3 bool) (TestSuite, func(
 		log,
 	)
 	require.NoError(t, err)
-	server := server.NewServer(host, 0, store, log, metrics.NoopMetrics)
+
+	svrCfg := &server.SvrConfig{
+		Host: host,
+		Port:       0,
```
[the remark this answers is not in the document]
Comment (epociask) [reply e2e/setup.go:129]: [5e038ec](https://github.com/Layr-Labs/eigenda-proxy/pull/76/commits/5e038ecf751ef2a94cb71439d93befb12fa03e86)

File: e2e/server_test.go [hunk 3]
```diff
@@ -108,7 +108,7 @@ func TestProxyClientWithLargeBlob(t *testing.T) {
 	}
 	daClient := client.New(cfg)
 	//  2MB blob
-	testPreimage := []byte(e2e.RandString(2000000))
+	testPreimage := []byte(e2e.RandString(4_000_000))
```
[the remark this answers is not in the document]
Comment (epociask) [reply e2e/server_test.go]: [5e038ec](https://github.com/Layr-Labs/eigenda-proxy/pull/76/commits/5e038ecf751ef2a94cb71439d93befb12fa03e86)

File: server/server.go (hunk 1 above)
Comment (epociask) [reply server/server.go:46]: I thought about this but felt it was unnecessary since the other cfg values are processed into embed struct fields

**Pull request closed without merging**